cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/edges/flows/EvaluationOrder.kt

```diff
@@ -91,3 +91,89 @@ class EvaluationOrders<NodeType : Node>(
         outgoing = outgoing,
     ),
     MirroredEdgeCollection<Node, EvaluationOrder>
+
+/**
+ * This function inserts the given [newNode] before the current node ([this]) in its existing EOG
+ * path.
+ *
+ * Before:
+ * ```
+ * <node1> -- EOG -->
+ *                    <node3>
+ * <node2> -- EOG -->
+ * ```
+ *
+ * We want to insert a new [EvaluationOrder] edge between all incoming edges of [this] (node3).
+ *
+ * Afterward:
+ * ```
+ * <node1> -- EOG -->
+ *                    <new node> -- EOG --> <node3>
+ * <node2> -- EOG -->
+ * ```
+ */
+fun Node.insertNodeBeforeInEOGPath(
+    newNode: Node,
+    builder: ((EvaluationOrder) -> Unit)? = null,
+): Boolean {
+    // Construct a new edge from the given node to the current node
+    val edge = EvaluationOrder(newNode, this).also(builder ?: {})
```

Comment:
Why don't you make the builder non-nullable instead and use `{}` as the default implementation?